test/e2e/specs/editor/plugins/block-hooks.spec.js

```diff
@@ -0,0 +1,292 @@
+				page,
+			} ) => {
+				await page.goto( `/?p=${ containerPost.id }` );
+				await expect(
+					page.locator( '.entry-content > *' )
+				).toHaveClass( [
+					'wp-block-heading',
+					getHookedBlockClassName( 'after', 'core/heading' ),
+					'dummy-paragraph',
+					getHookedBlockClassName( 'last_child', blockType ),
+				] );
+			} );
+
+			test( `should insert hooked blocks into ${ name } in editor and respect changes made there`, async ( {
+				admin,
+				editor,
+				page,
+			} ) => {
+				const expectedHookedBlockAfterHeading = {
+					name: 'core/paragraph',
+					attributes: {
+						className: getHookedBlockClassName(
+							'after',
+							'core/heading'
+						),
+					},
+				};
+
+				const expectedHookedBlockLastChild = {
+					name: 'core/paragraph',
+					attributes: {
+						className: getHookedBlockClassName(
+							'last_child',
+							blockType
+						),
+					},
+				};
+
+				await admin.editPost( postObject.id );
+				await expect
+					.poll( editor.getBlocks )
+					.toMatchObject( [
+						{ name: 'core/heading' },
+						expectedHookedBlockAfterHeading,
+						{ name: 'core/paragraph' },
+						expectedHookedBlockLastChild,
+					] );
+
+				const hookedBlock = editor.canvas.getByText(
+					getHookedBlockContent( 'last_child', blockType )
+				);
+				await editor.selectBlocks( hookedBlock );
+				await editor.clickBlockToolbarButton( 'Move up' );
+
+				// Save updated post.
+				const saveButton = page
+					.getByRole( 'region', { name: 'Editor top bar' } )
+					.getByRole( 'button', { name: 'Save', exact: true } );
+				await saveButton.click();
+				await page
+					.getByRole( 'button', { name: 'Dismiss this notice' } )
+					.filter( { hasText: 'updated' } )
+					.waitFor();
+
+				// Reload and verify that the new position of the hooked block has been persisted.
+				await page.reload();
+				await expect
+					.poll( editor.getBlocks )
+					.toMatchObject( [
+						{ name: 'core/heading' },
+						expectedHookedBlockAfterHeading,
+						expectedHookedBlockLastChild,
+						{ name: 'core/paragraph' },
+					] );
+
+				// Verify that the frontend reflects the changes made in the editor.
+				await page.goto( `/?p=${ containerPost.id }` );
+				await expect(
+					page.locator( '.entry-content > *' )
+				).toHaveClass( [
+					'wp-block-heading',
+					getHookedBlockClassName( 'after', 'core/heading' ),
+					getHookedBlockClassName( 'last_child', blockType ),
+					'dummy-paragraph',
+				] );
+			} );
+		} );
+	} );
+
+	test.describe( `Hooked blocks in Navigation Menu`, () => {
+		let postObject, containerPost;
+		test.beforeAll( async ( { requestUtils } ) => {
+			postObject = await requestUtils.createNavigationMenu( {
+				title: 'Navigation Menu',
+				status: 'publish',
+				content:
+					'<!-- wp:navigation-link {"label":"wordpress.org","url":"https://wordpress.org","kind":"custom"} /-->', // '<!-- wp:page-list /-->',
```

Comment:
Why is there a code comment with a different block included?

Reply:
These two failing tests might be related to that:

<img width="1308" alt="Screenshot 2025-02-14 at 11 36 47" src="https://github.com/user-attachments/assets/717826a3-8038-4eaf-bebc-63e06d6c2a65" />

Reply:
https://github.com/WordPress/gutenberg/pull/69044/commits/26c253930320695fa4bab5ba87cd4d18a3a0f393

Reply:
Hmm, still failing. Curiously, these tests have always been passing locally for me 🤔

Reply:
I figured it out. The Navigation block test was using a Page List block as one of the hooked blocks. However, there are no pages created in the test env, so that block came up empty.

Fixed in https://github.com/WordPress/gutenberg/pull/69044/commits/b8344051c8cf8bf7322e55fdec372226559d8fd4.